src/app/azureResource/actions.ts

```diff
@@ -0,0 +1,27 @@
+/***********************************************************
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License
+ **********************************************************/
+import actionCreatorFactory from 'typescript-fsa';
+import { SET } from '../constants/actionTypes';
+import { AzureResource } from './models/azureResource';
+
+export const AZURE_RESOURCES = 'AZURE_RESOURCES';
+export const BY_CONNECTION = '_CONNECTION';
+export const BY_HOSTNAME = '_HOST';
+
+const actionCreator = actionCreatorFactory(AZURE_RESOURCES);
+
+export interface SetActiveAzureResourceByConnectionStringActionParameters {
```

Comment:
is there any specific reason that we want to make 'hostName' a separate property? If we remove it, we can extract it out from connection string when we need it, so we don't need to pass it all the way down.